src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -41,18 +41,20 @@ public class Settings : ISettings
             new[] { "*.config" } :
             new[] { "*.Config", "*.config" };
 
-        public Dictionary<string, VirtualSettingSection> ComputedSections { get; }
+        private readonly Dictionary<string, VirtualSettingSection> _computedSections;
 
         public SettingSection GetSection(string sectionName)
         {
-            if (ComputedSections.TryGetValue(sectionName, out var section))
+            if (_computedSections.TryGetValue(sectionName, out var section))
             {
                 return section.Clone() as SettingSection;
             }
 
             return null;
         }
 
+        public IEnumerable<string> GetAllSettingSections() { return _computedSections.Keys; }
```

Comment:
`_computedSections` can be null here so we have to guard against that. We can do something like what `GetSection` does, with a lazy initialization of this dictionary if it hasn't already been initialized.

Reply:
How can it be null? Am I missing a constructor that fails to initialize it?

Reply:
I was talking with Martin and we also couldn't figure out when `_computedSections` would be null. It looks like the constructor for `_computedSections` initializes an empty dictionary, and from what I understand, the `Keys` property would also be empty, and not null.